tests/tree_algorithm.py

```diff
@@ -0,0 +1,176 @@
+                        if write_stdout_fail:
+                            fail(filestr + remark)
+                            print('')
+
+                        if write_csv_any:
+                            write_csv_line(
+                                csv_filepath, [file, str(i), str(j), False,
+                                               "{0:.3f}".format(end_isomorph - start_isomorph), "{0:.3f}".format(end_amount_isomorphisms - start_amount_isomorphisms),
+                                               "{0}".format(remark)]
+                            )
+                        error_adder = 1
+                    else:
+                        remark = "[PASS] Amount of isomorphisms is: " + str(amount_of_isomorphisms_actual)
+                        if write_csv_any:
+                            write_csv_line(
+                                csv_filepath, [file, str(i), str(j), True,
+                                               "{0:.3f}".format(end_isomorph - start_isomorph),
+                                               "{0:.3f}".format(end_amount_isomorphisms - start_amount_isomorphisms),
+                                               "{0}".format(remark)]
+                            )
+                        if write_stdout_passed:
+                            passed(filestr + remark)
+                            print('')
+                else:
+                    if are_isomorph_actual:
+                        remark = "[FAIL] Trees are not isomorphic, trees_are_isomorphic() did detect it to be isomorphic though."
+                        if write_stdout_fail:
+                            fail(filestr + remark)
+                        if write_csv_any:
+                            write_csv_line(
+                                csv_filepath, [file, str(i), str(j), False,
+                                               "{0:.3f}".format(end_isomorph - start_isomorph), "{0:.3f}".format(end_amount_isomorphisms - start_amount_isomorphisms),
+                                               "{0}".format(remark)]
+                            )
+                        error_adder = 1
+                    else:
+                        remark = "[PASS] Trees are not isomorphic."
+                        if write_csv_any:
+                            write_csv_line(
+                                csv_filepath, [file, str(i), str(j), True,
+                                               "{0:.3f}".format(end_isomorph - start_isomorph),
+                                               "{0:.3f}".format(end_amount_isomorphisms - start_amount_isomorphisms),
+                                               "{0}".format(remark)]
+                            )
+                        if write_stdout_passed:
+                            passed(filestr + remark)
+
+                error_count += error_adder
```

Comment:
Waarom is deze aparte error_adder nodig?

Reply:
Komt uit de unittests van Martijn. Dat is omdat als je voor elke combinatie van grafen maar 1 fout wil tellen als er iets fout gaat. Dus de error_adder wordt alleen maar op 1 gezet als het fout gaat en ook al wordt dat meerdere keren gedaan, gaat de error_count op het laatst alleen maar met 1 omhoog per combinatie van grafen.